rebase_automation.sh

```
@@ -0,0 +1,650 @@
+        if grep -q "$OLD_BUNDLE_VERSION" "$file" 2>/dev/null && ! grep -q "$NEW_BUNDLE_VERSION" "$file" 2>/dev/null; then
+            if safe_replace_version "$file" "$OLD_BUNDLE_VERSION" "$NEW_BUNDLE_VERSION" "bundle"; then
+                log_info "Updated bundle version in $file"
+                changed_files+=("$file")
+            fi
+        fi
+    done
+    
+    # Remove duplicates from changed_files array and report changes
+    if [[ ${#changed_files[@]} -gt 0 ]]; then
+        # Sort and remove duplicates
+        local unique_files=($(printf '%s\n' "${changed_files[@]}" | sort -u))
```

Shellcheck (0.11.0)

[warning] Prefer mapfile or read -a to split command output (or quote to avoid splitting).

(SC2207)